### PR TITLE
Fix go routine counter in tests

### PR DESCRIPTION
**Pull request by iulianpascalau**
- fixed go routines counter in tests by using a specialized component

### CODE REVIEW

File: testscommon/goroutines/testdata/test1.data [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+created by go.opencensus.io/stats/view.init.0
+	/home/user/Dev3/go/pkg/mod/go.opencensus.io@v0.23.0/stats/view/worker.go:34 +0x92
+
+goroutine 53 [select]:
+github.com/ElrondNetwork/elrond-go/trie.(*trieStorageManager).doProcessLoop(0xc0000f0000, {0x13248d8, 0xc000753100}, {0x1315fc8, 0x1ca7ca0}, {0x1315f98, 0xc0003993f8})
+	/home/user/Dev3/go/src/github.com/ElrondNetwork/elrond-go/trie/trieStorageManager.go:110 +0x11f
+github.com/ElrondNetwork/elrond-go/trie.(*trieStorageManager).doCheckpointsAndSnapshots(0xc000103040, {0x13248d8, 0xc000753100}, {0x1315fc8, 0x1ca7ca0}, {0x1315f98, 0xc0003993f8})
+	/home/user/Dev3/go/src/github.com/ElrondNetwork/elrond-go/trie/trieStorageManager.go:104 +0x3f
+created by github.com/ElrondNetwork/elrond-go/trie.NewTrieStorageManager
+	/home/user/Dev3/go/src/github.com/ElrondNetwork/elrond-go/trie/trieStorageManager.go:99 +0x5f7
```
Comment (BeniaminDrasovean) [inline testscommon/goroutines/testdata/test1.data:91]: Add empty line at the end of the file. Same for all .data files

**Review by BeniaminDrasovean — DISMISSED** [review]

File: testscommon/goroutines/testdata/test1.data (hunk 1 above)
Comment (iulianpascalau) [reply testscommon/goroutines/testdata/test1.data:91]: done

File: testscommon/goroutines/testdata/test3.data [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+created by github.com/libp2p/go-libp2p-asn-util.newIndirectAsnStore
+	/home/user/Dev3/go/pkg/mod/github.com/libp2p/go-libp2p-asn-util@v0.0.0-20200825225859-85005c6cf052/asn.go:87 +0xa5
+
+goroutine 53 [select]:
+github.com/ElrondNetwork/elrond-go/trie.(*trieStorageManager).doProcessLoop(0xc0000f0000, {0x13248d8, 0xc000753100}, {0x1315fc8, 0x1ca7ca0}, {0x1315f98, 0xc0003993f8})
+	/home/user/Dev3/go/src/github.com/ElrondNetwork/elrond-go/trie/trieStorageManager.go:110 +0x11f
+github.com/ElrondNetwork/elrond-go/trie.(*trieStorageManager).doCheckpointsAndSnapshots(0xc000103040, {0x13248d8, 0xc000753100}, {0x1315fc8, 0x1ca7ca0}, {0x1315f98, 0xc0003993f8})
+	/home/user/Dev3/go/src/github.com/ElrondNetwork/elrond-go/trie/trieStorageManager.go:104 +0x3f
+created by github.com/ElrondNetwork/elrond-go/trie.NewTrieStorageManager
+	/home/user/Dev3/go/src/github.com/ElrondNetwork/elrond-go/trie/trieStorageManager.go:99 +0x5f7
```
Comment (bogdan-rosianu) [inline testscommon/goroutines/testdata/test3.data:85]: empty line at the end of file? Also, can you remove `Dev3` from test data paths?

File: testscommon/goroutines/testdata/test2.data [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+created by go.opencensus.io/stats/view.init.0
+	/home/user/Dev3/go/pkg/mod/go.opencensus.io@v0.23.0/stats/view/worker.go:34 +0x92
+
+goroutine 53 [select]:
+github.com/ElrondNetwork/elrond-go/trie.(*trieStorageManager).doProcessLoop(0xc0000f0000, {0x13248d8, 0xc000753100}, {0x1315fc8, 0x1ca7ca0}, {0x1315f98, 0xc0003993f8})
+	/home/user/Dev3/go/src/github.com/ElrondNetwork/elrond-go/trie/trieStorageManager.go:110 +0x11f
+github.com/ElrondNetwork/elrond-go/trie.(*trieStorageManager).doCheckpointsAndSnapshots(0xc000103040, {0x13248d8, 0xc000753100}, {0x1315fc8, 0x1ca7ca0}, {0x1315f98, 0xc0003993f8})
+	/home/user/Dev3/go/src/github.com/ElrondNetwork/elrond-go/trie/trieStorageManager.go:104 +0x3f
+created by github.com/ElrondNetwork/elrond-go/trie.NewTrieStorageManager
+	/home/user/Dev3/go/src/github.com/ElrondNetwork/elrond-go/trie/trieStorageManager.go:99 +0x5f7
```
Comment (bogdan-rosianu) [inline testscommon/goroutines/testdata/test2.data:43]: empty line?

File: testscommon/goroutines/testdata/test1.data (hunk 1 above)
Comment (bogdan-rosianu) [inline testscommon/goroutines/testdata/test1.data:91]: empty line?

**Review by bogdan-rosianu — APPROVED** [review]

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**